packages/dds/tree/src/shared-tree/checkoutFlexTreeView.ts

```diff
@@ -30,9 +33,11 @@ export class CheckoutFlexTreeView<out TCheckout extends ITreeCheckout = ITreeChe
 	public readonly context: Context;
 
 	/**
-	 * Get a typed view of the tree content using the flex-tree API.
+	 * Get a view of the tree content using the flex-tree API.
 	 */
-	public readonly flexTree: FlexTreeField;
+	public get flexTree(): FlexTreeField {
```

Comment:
This is a functional change, fixing a case where root held onto by this context could get out of date in some cases that currently never happen, but could happen after some other changes (mainly if this context out lives a clear of the underlying context, causing it to recreate the root field.

Reply:
Removing of this cache however has caused a leak since we don't cache the root field anywhere else.

Reply:
Caching has been added to Context to cover this case.

The leak was detected by our leak unit tests for root access.